uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/controllers/sub-controllers/sick-request-ctrl_test.js

```diff
@@ -0,0 +1,163 @@
+(function (CRM) {
+  define([
+    'common/lodash',
+    'mocks/data/option-group-mock-data',
+    'common/angularMocks',
+    'leave-absences/shared/config',
+    'common/mocks/services/hr-settings-mock',
+    'mocks/apis/absence-period-api-mock',
+    'mocks/apis/absence-type-api-mock',
+    'mocks/apis/entitlement-api-mock',
+    'mocks/apis/work-pattern-api-mock',
+    'mocks/apis/leave-request-api-mock',
+    'mocks/apis/option-group-api-mock',
+    'mocks/apis/public-holiday-api-mock',
+    'common/mocks/services/api/contact-mock',
+    'leave-absences/shared/controllers/sub-controllers/sick-request-ctrl',
+    'leave-absences/shared/modules/shared-settings',
+  ], function (_, optionGroupMock) {
+    'use strict';
+
+    describe('SickRequestCtrl', function () {
+      var $log, $rootScope, $ctrl, modalInstanceSpy, $scope, $controller,
+        $provide, sharedSettings, AbsenceTypeAPI, SicknessRequestInstance,
+        date2016 = '01/12/2016';
+
+      beforeEach(module('leave-absences.templates', 'leave-absences.controllers',
+        'leave-absences.mocks', 'common.mocks', 'leave-absences.settings',
+        function (_$provide_) {
+          $provide = _$provide_;
+        }));
+
+      beforeEach(inject(function (_AbsencePeriodAPIMock_, _HR_settingsMock_,
+        _AbsenceTypeAPIMock_, _EntitlementAPIMock_, _WorkPatternAPIMock_,
+        _LeaveRequestAPIMock_, _OptionGroupAPIMock_, _PublicHolidayAPIMock_) {
+        $provide.value('AbsencePeriodAPI', _AbsencePeriodAPIMock_);
+        $provide.value('AbsenceTypeAPI', _AbsenceTypeAPIMock_);
+        $provide.value('EntitlementAPI', _EntitlementAPIMock_);
+        $provide.value('WorkPatternAPI', _WorkPatternAPIMock_);
+        $provide.value('HR_settings', _HR_settingsMock_);
+        $provide.value('LeaveRequestAPI', _LeaveRequestAPIMock_);
+        $provide.value('api.optionGroup', _OptionGroupAPIMock_);
+        $provide.value('PublicHolidayAPI', _PublicHolidayAPIMock_);
+      }));
+
+      beforeEach(inject(function (_$log_, _$controller_, _$rootScope_,
+        _AbsenceTypeAPI_, _SicknessRequestInstance_) {
+
+        $log = _$log_;
+        $rootScope = _$rootScope_;
+        $controller = _$controller_;
+        AbsenceTypeAPI = _AbsenceTypeAPI_;
+        SicknessRequestInstance = _SicknessRequestInstance_;
+        modalInstanceSpy = jasmine.createSpyObj('modalInstanceSpy', ['dismiss', 'close']);
+
+        spyOn($log, 'debug');
+        spyOn(AbsenceTypeAPI, 'all').and.callThrough();
+        spyOn(SicknessRequestInstance, 'init').and.callThrough();
+      }));
+
+      describe('sick request', function () {
```

Comment:
Can we add a test case to test the inheritance? Both for Sick and leave Controller
`expect(childController).toBe(jasmine.objectContaining(parentController))`